app/src/org/commcare/fragments/personalId/PersonalIdBiometricConfigFragment.java

```diff
@@ -124,50 +130,56 @@ private void reportAuthSuccess() {
+                requireContext(), biometricManager);
+        BiometricsHelper.ConfigurationStatus pinStatus = BiometricsHelper.checkPinStatus(requireContext(),
+                biometricManager);
+
+        String title="";
+        String message="";
         String fingerprintButton = null;
         String pinButton = null;
 
-        if (fingerprint == BiometricsHelper.ConfigurationStatus.Configured) {
-            title = getString(R.string.connect_verify_use_fingerprint_long);
-            message = getString(R.string.connect_verify_fingerprint_configured);
-            fingerprintButton = getString(R.string.connect_verify_agree);
-        } else if (pin == BiometricsHelper.ConfigurationStatus.Configured) {
-            title = getString(R.string.connect_verify_use_pin_long);
-            message = getString(R.string.connect_verify_pin_configured);
-            pinButton = getString(R.string.connect_verify_agree);
-            if (fingerprint == BiometricsHelper.ConfigurationStatus.NotConfigured) {
+        if(BIOMETRIC_TYPE == personalIdSessionDataViewModel.getPersonalIdSessionData().getRequiredLock()){
+            if(fingerprintStatus == BiometricsHelper.ConfigurationStatus.Configured) {  // Fingerprint is configured
+                title = getString(R.string.connect_verify_use_fingerprint_long);
+                message = getString(R.string.connect_verify_fingerprint_configured);
+                fingerprintButton = getString(R.string.connect_verify_agree);
+            }else{  // Need atleast Fingerprint configuration
+                title = getString(R.string.connect_verify_title);
+                message = getString(R.string.connect_verify_message);
                 fingerprintButton = getString(R.string.connect_verify_configure_fingerprint);
             }
-        } else {
-            title = getString(R.string.connect_verify_title);
-            message = getString(R.string.connect_verify_message);
-            if (fingerprint == BiometricsHelper.ConfigurationStatus.NotConfigured) {
+        }else if(PIN == personalIdSessionDataViewModel.getPersonalIdSessionData().getRequiredLock()){
+            if (fingerprintStatus == BiometricsHelper.ConfigurationStatus.Configured) { // Fingerprint is configured so all good
+                title = getString(R.string.connect_verify_use_fingerprint_long);
+                message = getString(R.string.connect_verify_fingerprint_configured);
+                fingerprintButton = getString(R.string.connect_verify_agree);
+            }else if (pinStatus == BiometricsHelper.ConfigurationStatus.Configured) {   // Only PIN is configure
+                title = getString(R.string.connect_verify_use_pin_long);
+                message = getString(R.string.connect_verify_pin_configured);
+                pinButton = getString(R.string.connect_verify_agree);
+                fingerprintButton = getString(R.string.connect_verify_configure_fingerprint);   // User can configure fingerprint
```

Comment:
this block seems indentical for the two auth types and should be abstracted to avoid code duplication.